src/transformers/models/bart/modeling_bart.py

```diff
@@ -1723,6 +1723,9 @@ def forward(
         )
         return_dict = return_dict if return_dict is not None else self.config.use_return_dict
 
+        if labels is not None:
```

Comment:
Hey @voidful,

Sorry this is not really what I meant. We should align the behavior of `BartForCausalLM` with `RobertaForCausalLM`. This change would mean that one can pass only `labels` and `input_ids` are created automatically - this is not how it's done in `GPT2LMHead`, `RobertaForCausalLM`, ... We're thinking a bit more on how to solve this internally with @patil-suraj - so we'll have to pause this PR for a bit

Reply:
It's OK. 
It may not be the best way and alleging the behavior of BartForCausalLM with RobertaForCausalLM should be a problem we need to deal with ?

I am confused with the design for token shift when calculating loss, input_ids and labels should be the same in xxxForCausalLM? 
or would it be better that moving label shift outside CausalLM and let the shifting determinate by users?